cost/aws/rightsize_ebs_volumes/aws_rightsize_ebs_volumes.pt

```diff
@@ -0,0 +1,1868 @@
+    if (result[key].length > 0) {
+      // Message for incident output
+      total_volumes = (ds_volumes_tag_filtered.length).toString()
+      total_unused_volumes = result[key].length.toString()
+      unused_volumes_percentage = "0%"
+      if (ds_volumes_tag_filtered.length > 0) {
+        unused_volumes_percentage = (result[key].length / ds_volumes_tag_filtered.length * 100).toFixed(2).toString() + '%'
+      }
+
+      recommended_action = "modification"
+      recommended_condition = "have had read/write activity below desired threshold"
+      if (key == "idle") {
+        recommended_action = "deletion"
+        recommended_condition = "have had zero read/write activity"
+      }
+
+      findings = [
+        "Out of ", total_volumes, " volumes analyzed, ",
+        total_unused_volumes, " (", unused_volumes_percentage, ") ",
+        "are ", key, " and recommended for ", recommended_action, ". "
+      ].join('')
+
+      days_noun = "day"
+      if (param_unused_days != 1) { days_noun += 's' }
+
+      if (param_include_status == 'Unattached Volumes') {
+        settings = [
+          "Volumes are considered ", key, " if they are unattached ",
+          "and have ", recommended_condition, " for at least ",
+          param_unused_days, " ", days_noun, ".\n\n"
+        ].join('')
+      }
+
+      if (param_include_status == 'Attached Volumes') {
+        settings = [
+          "Volumes are considered ", key, " if they are attached ",
+          "and have ", recommended_condition, " for at least ",
+          param_unused_days, " ", days_noun, ".\n\n"
+        ].join('')
+      }
+
+      if (param_include_status == 'All Volumes') {
+        settings = [
+          "Volumes are considered ", key, " if they ",
+          "have ", recommended_condition, " for at least ",
+          param_unused_days, " ", days_noun, ".\n\n"
+        ].join('')
+      }
+
+      disclaimer = "The above settings can be modified by editing the applied policy and changing the appropriate parameters."
+
+      summary_total_savings = ds_currency['symbol'] + ' ' + formatNumber(parseFloat(total_savings).toFixed(2), ds_currency['separator'])
+
+      // We need to return a list of objects for the resource count to be correct on the incident
+      // We also need to store a total savings and message for the whole list
+      //
+      // For now, using the first object in the list to store this,
+      // and these values are only used in the reports, not the exported data
+      result[key][0]['summary_message'] = findings + settings + disclaimer
+      result[key][0]['summary_total_savings'] = summary_total_savings
+      result[key][0]['total_savings'] = total_savings
+      result[key][0]['summary_policy_name'] = ds_applied_policy['name']
+    }
+  })
+
+  // Dummy items to ensure the policy's check statement always executes at least once
+  // BK attempting to remove these.. think they are causing NaN error using fpt retrieve_data
```

Comment:
I haven't really experienced this issue when using fpt with policies. Empty strings are valid non-null values. I think the NaN error is likely something else.

That said, these kinds of comments should be removed before putting a policy into the catalog

Reply:
Removed in [357ce0a](https://github.com/flexera-public/policy_templates/pull/2227/commits/357ce0a84b66c8c4a07283f6e788fd1c02f68cd5)